src/mqt/bench/cli.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import argparse
+
+from pytket import Circuit
+from pytket.qasm import circuit_to_qasm_str
+from qiskit import QuantumCircuit
+from qiskit.qasm2 import dumps as qiskit_circuit_to_str
+
+from mqt.bench import CompilerSettings, QiskitSettings, TKETSettings, get_benchmark
+
+
+def main() -> None:
+    """
+    Invoke :func:`get_benchmark` exactly once with the specified arguments.
+
+    The resulting QASM string is printed to the standard output stream;
+    no other output is generated.
+    """
+
+    parser = argparse.ArgumentParser(description="Generate a single benchmark")
+    parser.add_argument(
+        "--level",
+        type=str,
+        help='Level to generate benchmarks for ("alg", "indep", "nativegates" or "mapped")',
+        required=True,
+    )
+    parser.add_argument("--algorithm", type=str, help="Name of the benchmark", required=True)
+    parser.add_argument("--num-qubits", type=int, help="Number of Qubits")
```

Comment:
In the same spirit as above, this probably also has to be mandatory. Otherwise, it is not clear which benchmark to generate.
Note that there are certain non-scalable benchmarks that rather take a `str` that indicates the instance. You can see the special handling here: https://github.com/cda-tum/mqt-bench/blob/60e1fa7dad1584c3497384a9a906d64acc7f14a9/src/mqt/bench/benchmark_generator.py#L348-L379
I could see multiple different ways of handling that. Either is fine for me.

Reply:
Making this mandatory makes sense to me.
The benchmark instance thing is interesting. How about splitting those benchmarks up, at least from the user's POV, and removing the concept of instances from the interface? Should be feasible since neither shor nor groundstate parse the string in any way, and just have a fixed set of options.

Reply:
Just so that I get it right what you are proposing (which I think makes sense):
instead of having to specify `benchmark_name="shor"` and `benchmark_instance_name="xlarge"`, you would rather make this accessible as `benchmark_name="shor_xlarge"` for example?
Did I get that right? (in that case, these qubits wouldn't need a qubit number then)

Another special case in the benchmarks right now is that some benchmarks (grover and qwalk) have an additional parameter (how they handle ancillary qubits). In the UI, these are currently distinguished as `Grover's (no ancilla)` and `Grover's (v-chain)`. 
Any solution that solves the above, would also be applicable here.

Reply:
Yes, that's what I'm proposing (with nice display names in the web UI that make this obvious).

The ancilla handling seems to already work that way: `get_benchmark` does some special parsing of the given `benchmark_name` and then invokes `create_circuit` with the extra parameter.
I would prefer to completely get rid of this special case at this point, e.g.: Instead of loading python files by benchmark names, which requires one file per benchmark, have a `dict[benchmark_name, BenchmarkFactory]` (where `BenchmarkFactory` is akin to `(bench_name) -> QuantumCircuit`). Then, the same factory can be registered for multiple bench names, and the factory itself can be responsible for parsing the name.

Reply:
That sounds like a very reasonable change that would improve the library structure.
Might be a slightly bigger change though. How about "just making it work" here and doing the proposed refactor in a separate PR?

Reply:
Yes, absolutely agree 😃